From be9e2e3289d8ed66d3715f0d7b46dd404dd2e4a5 Mon Sep 17 00:00:00 2001
From: Patrick Mooney <pmooney@pfmooney.com>
Date: Fri, 7 Sep 2018 23:40:55 +0000
Subject: [PATCH] OS-7237 viona too persistent with legacy interrupt Reviewed
 by: Hans Rosenfeld <hans.rosenfeld@joyent.com> Reviewed by: Ryan Zezeski
 <rpz@joyent.com> Approved by: Ryan Zezeski <rpz@joyent.com>

---
 usr/src/cmd/bhyve/pci_virtio_viona.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/usr/src/cmd/bhyve/pci_virtio_viona.c b/usr/src/cmd/bhyve/pci_virtio_viona.c
index a671617258..26c8cdeeba 100644
--- a/usr/src/cmd/bhyve/pci_virtio_viona.c
+++ b/usr/src/cmd/bhyve/pci_virtio_viona.c
@@ -779,6 +779,9 @@ pci_viona_read(struct vmctx *ctx, int vcpu, struct pci_devinst *pi,
 		assert(size == 1);
 		value = sc->vsc_isr;
 		sc->vsc_isr = 0;	/* a read clears this flag */
+		if (value != 0) {
+			pci_lintr_deassert(pi);
+		}
 		break;
 	case VTCFG_R_CFGVEC:
 		assert(size == 2);
-- 
2.21.0

